tools/swift-inspect/Sources/swift-inspect/WindowsRemoteProcess.swift

```diff
@@ -233,37 +237,286 @@ internal final class WindowsRemoteProcess: RemoteProcess {
+    // The path to the dll assuming it's in the same directory as swift-inspect.
+    let swiftInspectPath = ProcessInfo.processInfo.arguments[0]
+    let dllPath: String =
+        URL(fileURLWithPath: swiftInspectPath).deletingLastPathComponent().path +
+        "\\SwiftInspectClient.dll"
+    var dllPathRemote = UnsafeMutableRawPointer(bitPattern: 0)
+    var dllPathUtf16Len: UInt = 0
+    var dllPathWritten = false
+    dllPath.withCString(encodedAs: UTF16.self) {
+      // Check that the dll file exists
+      let dwAttrib = GetFileAttributesW($0)
+      if dwAttrib == INVALID_FILE_ATTRIBUTES ||
+         Int32(dwAttrib) & FILE_ATTRIBUTE_DIRECTORY != 0 {
+        print("\(dllPath) doesn't exist")
+        return
+      }
+      // Allocate memory and write dllPath in the remote process
+      dllPathUtf16Len = UInt(wcslen($0) * MemoryLayout<WCHAR>.size)
+      dllPathRemote = VirtualAllocEx(self.process,
+                                     nil,
+                                     SIZE_T(dllPathUtf16Len + 1), // +1 for `\0`
+                                     DWORD(MEM_COMMIT),
+                                     DWORD(PAGE_READWRITE))
+      if dllPathRemote == LPVOID(bitPattern: 0) {
+        print("VirtualAllocEx failed \(GetLastError())")
+        return
+      }
+      if !WriteProcessMemory(self.process, dllPathRemote, $0,
+                             SIZE_T(dllPathUtf16Len), nil) {
+        print("WriteProcessMemory failed \(GetLastError())")
+        VirtualFreeEx(self.process, dllPathRemote, 0, DWORD(MEM_RELEASE))
+        return
+      }
+      dllPathWritten = true
+    }
+    if !dllPathWritten {
+      print("Failed to write dll path in the remote process")
+      return nil
+    }
+    return dllPathRemote
+  }
+
+  private func unloadDllAndPathRemote(
+      dwProcessId: DWORD, hKernel32: HMODULE!,
+      dllPathRemote: UnsafeMutableRawPointer) -> Bool {
+    // Get the dll module handle in the remote process to use it to
+    // unload it below.
+    // GetExitCodeThread returns a DWORD (32-bit) but the HMODULE
+    // returned from LoadLibraryW is a 64-bit pointer and may be truncated.
+    // So, search for it using the snapshot instead.
+    var hDllModule = HMODULE(bitPattern: 0)
+    let hModuleSnapshot: HANDLE =
+        CreateToolhelp32Snapshot(DWORD(TH32CS_SNAPMODULE), dwProcessId)
+    if hModuleSnapshot == INVALID_HANDLE_VALUE {
+      print("CreateToolhelp32Snapshot failed")
+      return false
+    }
+    var entry: MODULEENTRY32W = MODULEENTRY32W()
+    entry.dwSize = DWORD(MemoryLayout<MODULEENTRY32W>.size)
+    if !Module32FirstW(hModuleSnapshot, &entry) {
+      print("Module32FirstW failed")
+      return false
+    }
+    repeat {
+      let module: String = withUnsafePointer(to: entry.szModule) {
+        $0.withMemoryRebound(
+          to: WCHAR.self,
+          capacity: MemoryLayout.size(ofValue: $0) / MemoryLayout<WCHAR>.size) {
```

Comment:
This seems wrong?  This is going to be `sizeof(void *)` right?  Since you have an `UnsafeRawPointer` right?

Reply:
??? This looks to me like casting a char[] to wchar[] as szModule is a char[] that stores the name of the module. This follows what init? does above around line 171 above.

Reply:
I think that I had misread the value of `$0`.  Sorry!